spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
(Sorry for delay, was pretty busy)

> I'm adopting this design but before I want to let you know one thing. I also need to give Formula class a dataframe at the moment of parsing because I need to know whether a term is numerical or category while parsing.

Is it changing Formula behavior? How? Can't this behavior change be postponed until `process_dataframe`? 
Or, another interesting question: could the same formula be applied to _different_ dataframes (one having `a` as categorical and another having it numerical)? If it is, than "numericality" of variables should not be in Formula's state. If kind of variables is important on creation, than maybe something like this could possibly work:
```ruby
f1 = Formula.new('y~a+b', a: :numeric, b: :categorical)
f2 = Formula.from_df('y~a+b', df) # checks whether there are corresponding vectors + their type
```

I am afraid I'm not too familiar with this case, so many of my ideas are coming from generic "common reason", I'll need more details to guess the most sane architecture.